commands/create/kubeconfig/command.go

```diff
@@ -354,6 +359,40 @@ func createKubeconfigRunOutput(cmd *cobra.Command, cmdLineArgs []string) {
 	}
 }
 
+// getClusterDetails fetches cluster details to get the tenant cluster API endpoint,
+// and attempts first v5 and then falls back to v4.
+func getClusterDetails(clientWrapper *client.Wrapper, clusterID string, auxParams *client.AuxiliaryParams, verbose bool) (string, error) {
+	// Try v5 first, then fall back to v4.
+	if verbose {
+		fmt.Println(color.WhiteString("Fetching cluster details using the v5 API endpoint"))
+	}
+	clusterDetailsResponseV5, err := clientWrapper.GetClusterV5(clusterID, auxParams)
```

Comment:
With the latest version of this function, this could be clearer if there's:
```go
if err == nil {
    return clusterDetailsResponseV5.Payload.APIEndpoint, nil
}
```

So then you can remove one level indentation and focus on V4 rest of the function.

Reply:
Changed